src/main/java/com/verisign/storm/metrics/reporters/KafkaReporter.java

```diff
@@ -0,0 +1,149 @@
+ *
+ * See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+ */
+package com.verisign.storm.metrics.reporters;
+
+import com.google.common.base.Throwables;
+import com.verisign.ie.styx.avro.graphingMetrics.GraphingMetrics;
+import com.verisign.storm.metrics.graphite.GraphiteCodec;
+import com.verisign.storm.metrics.graphite.GraphiteConnectionFailureException;
+import kafka.javaapi.producer.Producer;
+import kafka.producer.KeyedMessage;
+import kafka.producer.ProducerConfig;
+import org.apache.avro.Schema;
+import org.apache.avro.io.*;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.avro.specific.SpecificRecordBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Properties;
+
+public class KafkaReporter extends AbstractReporter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaReporter.class);
+  public static final String KAFKA_TOPIC_NAME_FIELD = "metrics.kafka.topic";
+  public static final String KAFKA_BROKER_LIST_FIELD = "metadata.broker.list";
+
+  private String kafkaTopicName;
+  private String kafkaBrokerList;
+  private LinkedList<GraphingMetrics> buffer;
+
+  private Producer<String, byte[]> kafkaProducer;
+
+  private int failures;
+
+  public KafkaReporter(Map conf) {
+    super(conf);
+
+    if (conf.containsKey(KAFKA_TOPIC_NAME_FIELD)) {
+      kafkaTopicName = (String) conf.get(KAFKA_TOPIC_NAME_FIELD);
+    }
+    else {
+      throw new IllegalArgumentException("Field " + KAFKA_TOPIC_NAME_FIELD + " required.");
+    }
+
+    if (conf.containsKey(KAFKA_BROKER_LIST_FIELD)) {
+      kafkaBrokerList = (String) conf.get(KAFKA_BROKER_LIST_FIELD);
+    }
+    else {
+      throw new IllegalArgumentException("Field " + KAFKA_BROKER_LIST_FIELD + " required.");
+    }
+
+    Properties producerProps = new Properties();
+    for (String key : ((Map<String, Object>) conf).keySet()) {
+      producerProps.setProperty(key, (String) conf.get(key));
+    }
+
+    kafkaProducer = new Producer<String, byte[]>(new ProducerConfig(producerProps));
+    buffer = new LinkedList<GraphingMetrics>();
+    failures = 0;
+  }
+
+  @Override public void connect() throws GraphiteConnectionFailureException {
+    return;
+  }
+
+  @Override public void disconnect() throws GraphiteConnectionFailureException {
+    return;
+  }
+
+  @Override public void appendToBuffer(String prefix, Map<String, Object> metrics, long timestamp) {
+    Map<String, Double> metricsDoubleMap = new HashMap<String, Double>();
+
+    for (String key : metrics.keySet()) {
+      try {
+        Double value = Double.parseDouble(GraphiteCodec.format(metrics.get(key)));
```

Comment:
There is a possibility that we could run into a NPE here, in the case where `metrics.get(key)` returns `null` as the actual value associated with the key in the map.  Should we guard against this?